File: src/examples/addSignatories/basic.cs
using Criipto.Signatures;
using Criipto.Signatures.Models;

using (var client = new CriiptoSignaturesClient("{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}")) {
    // Setup document input
    var documents = new List<DocumentInput>{
        new DocumentInput {
            pdf = new PadesDocumentInput {
                title = "Dotnet Sample",
                blob = new Byte[64], // Should be the bytes of a PDF file
                storageMode = DocumentStorageMode.Temporary
            }
        }
    };

    // Setup signature order input
    var createSignatureOrderInput = new CreateSignatureOrderInput
    {
        title = "Dotnet Sample",
        documents = documents
    };

    // Create signature order
    var signatureOrder = await client.CreateSignatureOrder(createSignatureOrderInput);

    // Add signatories
    var signatories = await client.AddSignatories(signatureOrder.id, new List<CreateSignatureOrderSignatoryInput>() {
        new CreateSignatureOrderSignatoryInput() {
            reference = "A",
            documents = signatureOrder.documents.Select(d => new SignatoryDocumentInput() {
                id = d.id,
                preapproved = true
            }).ToList()
        },
        new CreateSignatureOrderSignatoryInput() {
            reference = "B",
            evidenceValidation = new List<SignatoryEvidenceValidationInput>() {
                new SignatoryEvidenceValidationInput() {
                    key = "cprNumberIdentifier",
                    value = "11223344-5555"
                }
            }
        }
    });

    foreach (var signatory in signatories) {
        Console.WriteLine(signatory.href); // Signing link, redirect user to this link, or send it in an email
    }
}

File: src/examples/addSignatories/basic.py
from criipto_signatures import CriiptoSignaturesSDKSync
from criipto_signatures.models import (
  CreateSignatureOrderInput,
  DocumentInput,
  PadesDocumentInput,
  DocumentStorageMode,
  AddSignatoriesInput,
  CreateSignatureOrderSignatoryInput,
  SignatoryDocumentInput,
  SignatoryEvidenceValidationInput,
)

client = CriiptoSignaturesSDKSync(
  "{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"
)

# Create signature order
signatureOrder = client.createSignatureOrder(
  CreateSignatureOrderInput(
    title="Python sample",
    documents=[
      DocumentInput(
        pdf=PadesDocumentInput(
          title="My document",
          blob=bytes("...", "utf-8"),  # bytes object, or a base64 encoded string
          storageMode=DocumentStorageMode.Temporary,
        )
      )
    ],
  )
)

# Add signatories
signatories = client.addSignatories(
  AddSignatoriesInput(
    signatureOrderId=signatureOrder.id,
    signatories=[
      CreateSignatureOrderSignatoryInput(
        reference="A",
        documents=[
          SignatoryDocumentInput(id=document.id, preapproved=True)
          for document in signatureOrder.documents
        ],
      ),
      CreateSignatureOrderSignatoryInput(
        reference="B",
        evidenceValidation=[
          SignatoryEvidenceValidationInput(
            key="cprNumberIdentifier", value="112233445555"
          )
        ],
      ),
    ],
  )
)

for signatory in signatories:
  print(
    signatory.href
  )  # Signing link, redirect user to this link, or send it in an email
